ssm/message_directory.py

```diff
@@ -0,0 +1,138 @@
+"""
+Copyright (C) 2018 STFC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+@author: Greg Corbett
+"""
+
+import logging
+import os
+import uuid
+
+# logging configuration
+LOG = logging.getLogger(__name__)
+
+
+class MessageDirectory(object):
+    """A structure for holding Accounting messages in a directory."""
+
+    def __init__(self, path):
+        """Create a new directory structure for holding Accounting messages."""
+        self.directory_path = path
+
+    def add(self, data):
+        """Add the passed data to a new file and return it's name."""
+        # Use uuid4 to create a file name because uuid4 is
+        # sufficient to create an unique ID.
+        name = uuid.uuid4()
+
+        try:
+            # Open the file and write the provided data into the file.
+            with open("%s/%s" % (self.directory_path, name), 'w') as message:
+                message.write(data)
+        except (IOError, OSError) as error:
+            LOG.error("Could not create file %s/%s: %s",
```

Comment:
looking at this with fresh eyes, just logging the error seems like a bad idea. I don't think the add method should return a filename if a file was not created.

Thoughts @tofu-rocketry?

I can't quite parse what the dirq module would do in this case, see [here](https://github.com/cern-mig/python-dirq/blob/75f63e193c209edd40a7a643386100d953bcbdab/dirq/QueueSimple.py#L212) and [here](https://github.com/cern-mig/python-dirq/blob/75f63e193c209edd40a7a643386100d953bcbdab/dirq/QueueSimple.py#L178). Does it raise an OSError?

Reply:
dirq loops until it can create the directory or an error other than it already exists is raised.

Prob just raise the error.